### ISSUE TITLE
IPsec VTI - Potential lockout risk if policies isn't unchecked and configuration is applied

### ISSUE THREAD

**Issue by Monviech** [opnsense/core#6936]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [x] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [x] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Is your feature request related to a problem? Please describe.**

When creating a VTI IPsec tunnel, there can be a fatal user error that will install a 0.0.0.0/0 ::/0 ipsec kernel policy. It happens when "policies" in a connection "children" isn't deactivated and then applying the configuration.

Afterwards, the firewall won't respond to any network traffic and you have to go into the local terminal to stop the strongswan service. Even a reboot won't help in that case, since strongswan will automatically load all configuration after boot.

**Describe the solution you like**

There could be an additional popup window or a warning when a "reqid" is explicitly set in the children. A user would probably only set a static reqid if they have created a VTI interface beforehand and plan on using a routed IPsec tunnel. The popup could warn them about the potential lockout if policies isn't unchecked and the configuration is applied.

**Describe alternatives you considered**

Changing the default of "policies" to "0". Then a user would have to explicitly enable policies when they want to use them. It would reduce the risk of a total lockout. But it would probably cause confusion for users who just want to create a simple policy based tunnel.

**Comment by mimugmail** [comment]
Isn't this also the case with legacy implementation?

**Comment by AdSchellevis** [comment]
I think we forgot to require a policy in this case, which makes it a bit too easy to trip over it.

**Comment by AdSchellevis** [comment]
@Monviech I'm not sure I do understand the issue correctly, while reading https://docs.strongswan.org/docs/5.9/swanctl/swanctlConf.html I would expect if we keep `local_ts` and `remote_ts ` empty the tunnel endpoint addresses are used (which should be more or less safe).

**Comment by Monviech** [comment]
@AdSchellevis 
I was meaning this as reference, the warning here:
https://docs.opnsense.org/manual/how-tos/ipsec-s2s-conn-route.html#children

I have managed to lock myself out one time yesterday in my test setup and had to stop the strongswan daemon.

Right now, I tried my best to replicate it with all kinds of different wrong inputs and I can't replicate it anymore. If I try my hardest and can't break it I guess there is no issue and it was only a fluke.

**Comment by AdSchellevis** [comment]
Unfortunately adding a warning requires javascript on the frontend part, I'm not against it, but it's not high on my priority list. Preventing an exact match from happening with a validation is easier, but still leaves options to kill your setup (overlaps of any kind are problematic)

**Comment by Monviech** [comment]
Thanks for your time, I will close this as I can't replicate it anymore.

**Comment by WMP** [comment]
# IPsec `0.0.0.0/0` selectors (policies enabled) blackhole the firewall's own loopback

- **Area:** VPN → IPsec (Connections / policy-based Phase 2, strongSwan)
- **Version:** OPNsense **Business** `26.4.1p1_3` (type `opnsense-business`)
- **Type:** bug — silent self-lockout, no warning

## TL;DR

A child with `0.0.0.0/0 ↔ 0.0.0.0/0` and **Policies enabled** installs an SPD that matches `127.0.0.1`. The firewall tunnels its own loopback → local IPv4 dies. IPv6 survives. No warning. `pf`'s `set skip on lo0` does not help (SPD ≠ pf).

## Reproduce

```sh
# VPN → IPsec → Connections → Child:
#   Mode = Tunnel, Local = 0.0.0.0/0, Remote = 0.0.0.0/0, Policies = ENABLED
# apply, bring tunnel up, then:
ping -c3 127.0.0.1
```

```
3 packets transmitted, 0 packets received, 100.0% packet loss
```

## Symptom map (IPv4 dead, IPv6 fine)

```sh
ping  -c3 127.0.0.1     # 100% loss
ping6 -c3 ::1           # 0% loss   <-- policy is IPv4-only
```

## It is NOT pf — loopback is skipped

```sh
pfctl -vsI | grep -iB1 skip
```
```
lo0 (skip)
pfsync0 (skip)
```

## The loopback packet leaves via `enc0`, never `lo0`

```sh
# session A
tcpdump -ni lo0                         # 0 IPv4 packets during ping; DOES catch ::1
# session B
ping -c2 127.0.0.1
```

```sh
tcpdump -nei pflog0 icmp and host 127.0.0.1
```
```
rule 122/0(match): pass out on enc0: 127.0.0.1 > 127.0.0.1: ICMP echo request, id 37429, seq 0
```
> `@122` is the stock auto `pass out ... on enc0` rule — a symptom, not the cause.

## The cause — SPD matches all IPv4

```sh
setkey -DP
```
```
0.0.0.0/0[any] 0.0.0.0/0[any] any
        in ipsec
        esp/tunnel/198.51.100.20-203.0.113.10/unique:1
0.0.0.0/0[any] 0.0.0.0/0[any] any
        out ipsec
        esp/tunnel/203.0.113.10-198.51.100.20/unique:1
```

`127.0.0.1 ∈ 0.0.0.0/0` → looped-back traffic is diverted into the tunnel.

## Not just ICMP — every local service on loopback

```sh
sockstat -4l | grep 127.0.0.1
```
```
root     lighttpd  tcp4  127.0.0.1:443     # GUI back-end
root     lighttpd  tcp4  127.0.0.1:80
unbound  unbound   tcp4  127.0.0.1:953     # local resolver control
```
```sh
fetch -qo- http://127.0.0.1/          # hangs
```

Kernel counters confirm the drop + upstream side effect:

```sh
netstat -sp ip   | grep -i 'no route'      # -> NNNN output packets discarded due to no route
netstat -sp icmp | grep -i unreachable     # -> destination unreachable: NNNN
pfctl -ss | grep icmp                       # gateway monitor blackholed:
# all icmp 203.0.113.10:28051 -> 203.0.113.1:8   0:0   <-- dpinger probes tunneled
```

## Expected

Either a bypass, a warning, or both.

**(a) auto bypass SPD** for loopback/host-local, higher priority than user tunnels — e.g.:

```sh
# what the box SHOULD install so 0.0.0.0/0 can never eat loopback:
setkey -c <<'EOF'
spdadd 127.0.0.0/8 127.0.0.0/8 any -P out none;
spdadd 127.0.0.0/8 127.0.0.0/8 any -P in  none;
EOF
```

**(b) GUI warning** when `0.0.0.0/0` + policies enabled:

```
⚠ Traffic selector 0.0.0.0/0 with policies enabled captures ALL traffic
  originated by this firewall, including loopback (127.0.0.1) and local
  services (DNS, GUI). For full-tunnel/route-based use VTI with policies
  DISABLED.
```

## Bad vs good config

```diff
  # Child (VPN → IPsec → Connections)
  Mode:    Tunnel
  Local:   0.0.0.0/0
  Remote:  0.0.0.0/0
- Policies: [x] enabled      # policy-based 0.0.0.0/0 -> eats loopback
+ Policies: [ ] disabled     # route-based/VTI -> routing decides, loopback safe
+ Reqid:    1                # must match the VTI interface
```

Policy-based alternative — never `0.0.0.0/0`, narrow it:

```diff
- Local: 0.0.0.0/0   Remote: 0.0.0.0/0
+ Local: 172.29.0.0/16   Remote: 172.19.12.0/22    # loopback no longer matches
```

## Verify the fix

```sh
setkey -DP | grep -c '0.0.0.0/0'   # expect 0 (VTI) or narrowed prefixes (policy)
ping -c3 127.0.0.1                  # expect 0% loss
fetch -qo- http://127.0.0.1/ >/dev/null && echo loopback-ok
```

## Workarounds

- **Route-based/VTI:** Policies **disabled**, keep `0.0.0.0/0`, set fixed `reqid` = VTI iface, route via static/FRR.
- **Policy-based:** narrow selectors to real subnets, never `0.0.0.0/0`.

## Related (different mechanism, same `0.0.0.0/0` family)

A separate `0.0.0.0/0` VTI foot-gun reported on r/opnsense ([IPsec VTI bug](https://www.reddit.com/r/opnsense/comments/1quwi7d/ipsec_vti_bug/), OPNsense 25.7.9): adding the VTI as a **gateway** (esp. with "Far gateway") injects a spurious **default route** cost 0 via the peer VTI IP, overriding BGP.

| | This report | r/opnsense thread |
|---|---|---|
| Layer | IPsec **SPD** (`setkey -DP`) | **routing table** (`netstat -rn`) |
| Trigger | `0.0.0.0/0` child, **policies enabled** | VTI added as gateway / "Far gateway" |
| Breaks | firewall's own loopback + local services | BGP / all connectivity via bad default |
| Seen in | `setkey -DP` shows `0.0.0.0/0 ... ipsec` | `netstat -rn` shows `default ... via <VTI peer>` |
| Fix | bypass SPD / disable policies / narrow TS | route via FRR, don't add VTI as gateway |

Both point at the same gap: `0.0.0.0/0` in a VTI context silently captures traffic it should not, with no warning.

## Refs

- https://docs.opnsense.org/manual/how-tos/ipsec-s2s-conn-route.html
- https://docs.opnsense.org/manual/how-tos/ipsec-s2s-route.html
